test/integration/node-specific/auto_connect.test.ts

```diff
@@ -0,0 +1,689 @@
+    });
+  });
+
+  // GridFS APIs are all made up of CRUD APIs on collections and dbs.
+
+  context(`class ListCollectionsCursor`, () => {
+    it(`#forEach()`, async () => {
+      const collections = client.db().listCollections();
+      await collections.forEach(item => {
+        expect(item).is.an('object');
+      });
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`#hasNext()`, async () => {
+      const collections = client.db().listCollections();
+      await collections.hasNext();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`#next()`, async () => {
+      const collections = client.db().listCollections();
+      await collections.next();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`#toArray()`, async () => {
+      const collections = client.db().listCollections();
+      await collections.toArray();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`#tryNext()`, async () => {
+      const collections = client.db().listCollections();
+      await collections.tryNext();
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+  });
+
+  context(`class ListIndexesCursor`, () => {
+    it(`#forEach()`, async () => {
+      const indexes = client.db().collection('test').listIndexes();
+      await indexes
+        .forEach(item => {
+          expect(item).is.an('object');
+        })
+        .catch(() => null);
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`#hasNext()`, async () => {
+      const indexes = client.db().collection('test').listIndexes();
+      await indexes.hasNext().catch(() => null);
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`#next()`, async () => {
+      const indexes = client.db().collection('test').listIndexes();
+      await indexes.next().catch(() => null);
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`#toArray()`, async () => {
+      const indexes = client.db().collection('test').listIndexes();
+      await indexes.toArray().catch(() => null);
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+
+    it(`#tryNext()`, async () => {
+      const indexes = client.db().collection('test').listIndexes();
+      await indexes.tryNext().catch(() => null);
+      expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+    });
+  });
+
+  context(`class MongoClient`, () => {
+    it(`#withSession()`, async () => {
+      await client.withSession(async session => {
+        expect(session).to.be.instanceOf(ClientSession);
+      });
+      expect(client).to.not.have.property('topology'); // withSession won't connect, that's expected
```

Comment:
I glanced through this PR and I don't have much to add, but I don't think we should make it a convention to use the `it` block for anything other than describing an action. The idea behind the test conventions is to make it easy to tell the intention from reading the actual spec printout. This comment here wouldn't show in the printout. We'd have Mongo Client Autoconnect class MongoClient #withSession()" which tells us nothing. If we're going for clarity, the #tag method names should be in the context or describe blocks and the `it` blocks should actually state the expectation.